app/assets/javascripts/Components/submission_table.jsx

```diff
@@ -348,12 +350,70 @@ class RawSubmissionTable extends React.Component {
 
         received: data => {
           // Called when there's incoming data on the websocket for this channel
-          this.fetchData();
+          if (data["status"] != null) {
+            let message_data = this.generateFlashMessageContentsUsingStatus(data);
+            this.renderFlashMessages(message_data);
+          }
+          if (data["update_table"] != null) {
+            this.fetchData();
+          }
         },
       }
     );
   };
 
+  generateFlashMessageContentsUsingStatus = status_data => {
+    let message_data = {};
+
+    switch (status_data["status"]) {
+      case "failed":
+        if (!status_data["exception"] || !status_data["exception"]["message"]) {
+          message_data["error"] = "Failed";
+        } else {
+          message_data["error"] = status_data["exception"]["message"];
+        }
+        break;
+      case "completed":
+        message_data["success"] = "Completed";
+        break;
+      case "queued":
+        message_data["notice"] = "Queued";
+        break;
+      default:
+        let progress = status_data["progress"];
+        let total = status_data["total"];
+        message_data["notice"] = `${progress}/${total} Submissions collected`;
+    }
+    if (status_data["warning_message"]) {
+      message_data["warning"] = status_data["warning_message"];
+    }
+    return message_data;
+  };
+  renderFlashMessages = message_data => {
```

Comment:
You've done a good job extracting the relevant parts. You should move these functions into a `flash.js` file and import them from that file. Ideally also modify `ajax_events.js` to import at least the shared constants (`FLASH_KEYS`) from that file.